keras_eval/eval.py

```diff
@@ -176,28 +175,29 @@ def save_probabilities_labels(self, id, save_path):
 
     def compute_inference_probabilities(self, concept_dictionary, probabilities):
         '''
-        Combines probabilities based on key "group" in concept_dictionary and saves the values in self.probabilities
+        Computes the probability inference based on key "group" in concept_dictionary and saves the values in self.probabilities
 
         Args:
             concept_dictionary: It is the dictionary which contains all the granular concepts and the mapping with the groups.
             probabilities: These are computed granular probabilities
 
         '''
+        self.combined_probabilities = utils.combine_probabilities(probabilities, self.combination_mode)
 
         for concept in concept_dictionary:
-
             if concept['group'] in self.group_id_dict.keys():
                 self.group_id_dict[concept['group']].append(concept['class_index'])
             else:
                 self.group_id_dict[concept['group']] = [concept['class_index']]
 
-        self.combined_probs = [[[0.0, ] * len(self.concept_labels)] * len(probabilities[0])]
-        self.combined_probs = np.array(self.combined_probs)
+        inference_probabilities = np.zeros((1, len(self.combined_probabilities), len(self.concept_labels)))
```

Comment:
why is the shape assigned this way? what is the 1 here?
Also why not self.combined_probabilities.shape

Reply:
I think it should be computed for every model independently to keep consistency with the previous definitions.

Reply:
I decided to `utils.combine_probabilities` all the model probabilities at the very beginning of this function to simplify the inference, that's why the first index of `inference_probabilities` is 1. 

I kind of see your point, @jsalbert . I can change that.

Reply:
If that was the case why adding an extra index